### PR TITLE
Expose syntheticNamedExports to ModuleInfo

### PR DESCRIPTION
**Pull request by Amareis**
This PR contains:
- [ ] bugfix
- [x] feature
- [ ] refactor
- [x] documentation
- [ ] other

Are tests included?
- [x] yes (*bugfixes and features will not be merged without tests*)
- [ ] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers:

### Description

In [docs](https://rollupjs.org/guide/en/#load) there is line about `meta` and `syntheticNamedExports` in `ModuleInfo`:
image

### CODE REVIEW

File: src/ExternalModule.ts [hunk 1]
```diff
@@ -58,7 +58,8 @@ export default class ExternalModule {
 			},
 			isEntry: false,
 			isExternal: true,
-			meta
+			meta,
+			syntheticNamedExports: false
```
Comment (Amareis) [inline src/ExternalModule.ts:62]: I wonder if we can somehow allow to use `syntheticNamedExports` with external modules, it will be great!

Comment (lukastaegert) [reply src/ExternalModule.ts:62]: It would not work the same way as for internal modules. For internal modules, syntheticNamedExports are only used for named exports that cannot be resolved. For external modules, we do not know which exports can or cannot be resolved. So the question is how to go about this. Either just make all named exports "synthetic" or implement a runtime detection logic. And how would this interact with the runtime detection logic for `__esModule`? If you want to get missing named exports from the default export, should it be the exported namespace or the `namespace.default` property? (In real life use-cases, likely the former).

And for formats other than ESM, you can already import whatever you want from external dependencies without errors being thrown.

In short, these are the reasons why we do not have this at the moment.

File: src/Module.ts [hunk 2]
```diff
@@ -284,7 +284,11 @@ export default class Module {
 			},
 			isEntry,
 			isExternal: false,
-			meta
+			meta,
+			get syntheticNamedExports() {
+				//since it's public and not readonly, we should use getter to ensure value is always actual
+				return module.syntheticNamedExports;
```
Comment (lukastaegert) [inline src/Module.ts:290]: Actually I would prefer to follow the route of `hasModuleSideEffects` here which makes this mutable object the source of truth for this property (even though it is public) as it helps reduce internal complexity. I.e. remove `syntheticNamedExports` from `Module` in favour of `info.syntheticNamedExports`. At the very least, it would be consistent.

**Review by lukastaegert — CHANGES_REQUESTED** [review]
Thanks! I would consider aligning the handling of the property with moduleSideEffects and meta, though.

File: src/Module.ts (hunk 2 above)
Comment (Amareis) [reply src/Module.ts]: Done!

**Review by lukastaegert — APPROVED** [review]
Looks great, thanks a lot!
